Make the Null Dao as default as H2 is not a production ready database and have a few CVEs (#676)
jenkinsci/pipeline-maven-plugin: README.adoc

**For tests purposes only**, pre-releases (beta) versions are available in the https://updates.jenkins.io/#experimental-plugin-site[Experimental Plugin Site] and development versions (incremental releases) are archived on every successful build on https://ci.jenkins.io/job/Plugins/job/pipeline-maven-plugin/[our CI environment].

> CAUTION: **If you want to use the feature <<feature-trigger-downstream>> it is critical to <<db-setup,setup a database>>** as per default it's not anymore H2.

Reviewer: ```suggestion
CAUTION: **If you want to use the feature <<feature-trigger-downstream>> it is critical to <<db-setup,setup a database>>** as H2 is not preconfigured by default any more.
```

[source,groovy]
----
pipeline {
  agent any
  stages {
    stage("Build") {
      steps {
        git url: 'https://github.com/cyrille-leclerc/multi-module-maven-project'

        withMaven(
            // Maven installation declared in the Jenkins "Global Tool Configuration"
            maven: 'maven-3', // <1>
            // Use `$WORKSPACE/.repository` for local repository folder to avoid shared repositories
            mavenLocalRepo: '.repository', // <2>
            // Maven settings.xml file defined with the Jenkins Config File Provider Plugin
            // We recommend to define Maven settings.xml globally at the folder level using
            // navigating to the folder configuration in the section "Pipeline Maven Configuration / Override global Maven configuration"
            // or globally to the entire master navigating to  "Manage Jenkins / Global Tools Configuration"
            mavenSettingsConfig: 'my-maven-settings' // <3>
        ) {

          // Run the maven build
          sh "mvn clean verify"

        } // withMaven will discover the generated Maven artifacts, JUnit Surefire & FailSafe & FindBugs & SpotBugs reports...

>     }

Author: FTR for some reasons it looks this change have been copied to master already via https://github.com/jenkinsci/pipeline-maven-plugin/commit/2c46922112293e7d39ae685e36db395579ef9632

    }
  }
}

pipeline {
  agent any
  stages {
    stage("Build") {
      steps {
        withMaven(traceability: true){
            .....
        }
      }
    }
  }
}

> 

Author: FTR for some reasons it looks this change have been copied to master already via https://github.com/jenkinsci/pipeline-maven-plugin/commit/2c46922112293e7d39ae685e36db395579ef9632
